vispy/app/tests/test_canvas.py

```diff
@@ -103,18 +103,18 @@ def on_draw(ev):
             gloo.clear('black')
             gloo.set_viewport(0, 0, *c.physical_size)
             image1.draw()
-            
+
         rgba_result = c.render()
         assert not np.allclose(rgba_result[..., :3], 0)
 
 
 @requires_application()
 @pytest.mark.parametrize("mouse_event_type, button, buttons, expected_button, expected_buttons", [
-    ('mouse_press', 1, [], 1, [1]),
-    ('mouse_release', 1, [1], 1, [1]),
+    ('mouse_press', 1, [1], 1, [1]),
+    ('mouse_release', 1, [2], 1, [2]),
     # left click pressed and held, followed by a right click
-    ('mouse_press', 2, [1], 2, [1, 2]),
-    ('mouse_release', 2, [1, 2], 2, [1, 2]),
+    ('mouse_press', 2, [1, 2], 2, [1, 2]),
+    ('mouse_release', 2, [1], 2, [1]),
 ])
```

Comment:
I wasn't thinking anything too complicated, mostly just adding some test cases here. I wanna make sure that we don't break this new behaviour in the future, as well as ensuring this actually behaves as you were expending on all our tested backends!